src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -117,11 +117,14 @@ internal void EnsureInitialized(WebHostSettings settings)
         {
             lock (_syncLock)
             {
+                // Determine whether we should do normal or standby initialization
                 if (!WebScriptHostManager.InStandbyMode)
                 {
-                    // standby mode can only change from true to false
-                    // when standby mode changes, we reset all instances
-                    if (_activeHostManager == null)
+                    // We're not in standby mode. There are two cases to consider:
+                    // 1) We _were_ in standby mode and now we're ready to specialize
+                    // 2) We're doing non-specialization normal initialization
+                    if (_activeHostManager == null && 
+                        (_standbyHostManager == null || _settingsManager.ContainerReady))
```

Comment:
This new line captures the above cases - either we weren't in standby (_standbyHostManager == null) or we were (_standbyHostManager != null) in which case we wait on ContainerReady to be true